config/allows_pooling.go

```diff
@@ -0,0 +1,28 @@
+package config
+
+type allowsPoolingOption bool
+
+func WithAllowsPooling(allowsPooling bool) interface {
+	HTTPEndpointOption
+	TCPEndpointOption
+	TLSEndpointOption
+	LabeledTunnelOption
```

Comment:
Have to do it for labeled tunnel too because common opts... we can have a different struct which is with is `allButLabeledCommon` if you prefer. Or I can just break it out for the 3.

We did it this way for description for agent endpoints for example

Reply:
Just because it's in the `CommonOpts` doesn't mean we have to expose the `LabeledTunnelOption` interface. We can just let it be a dead option for labeled tunnels.

```suggestion
```

Reply:
There was a situation where leaving an option undefined broke labeled tunnels with description... I don't recall what so I'll look.

Reply:
Okay I figured out what happened:

https://github.com/ngrok-private/ngrok/blob/main/go/cmd/ngrok/tung/config.go#L57

We do this for every tunnel, including LabeledTunnels... as long as we don't call code this way for allows_pooling it'll be fine.

Reply:
Yeah, let's just do it the right way from the agent side so we don't have to pretend to support pooling with labeled tunnels.